sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1188,6 +1188,11 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(i) if sqlContext.conf.supportQuotedRegexColumnName =>
+      Column(UnresolvedRegex(i, None))
+    case ParserUtils.qualifiedEscapedIdentifier(i, j)
+      if sqlContext.conf.supportQuotedRegexColumnName =>
```

Comment:
Nit: style issue. We prefer to add the extra two space before `if` in this case.